### PR TITLE
Emotion. Styled Components can be used as selectors

### PR DESCRIPTION
**Pull request by sgrishchenko**
cc @turansky @aerialist7 @stefanthaler

### CODE REVIEW

File: kotlin-emotion/src/main/kotlin/emotion/styled/styled.ext.kt [hunk 1]
```diff
@@ -14,12 +14,18 @@ private inline val Props.theme: Theme
 fun <P : PropsWithClassName> ElementType<P>.styled(
     options: (StyledOptions.() -> Unit)? = null,
```
Comment (turansky) [inline kotlin-emotion/src/main/kotlin/emotion/styled/styled.ext.kt:15]: Redundant?

File: kotlin-emotion/src/main/kotlin/emotion/styled/generateId.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package emotion.styled
+
+import emotion.hash.hash
+import react.ElementType
+
+private val identifiers = mutableMapOf<String, Int?>()
+
+internal fun generateId(component: ElementType<*>): String {
+    val displayName: String? = component.asDynamic().displayName
+    val name = displayName?.let { escape(it) } ?: "Anonymous"
```
Comment (turansky) [inline kotlin-emotion/src/main/kotlin/emotion/styled/generateId.kt:10]: Anonymous for intrinsic?

File: kotlin-emotion/src/main/kotlin/emotion/hash/hash.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+@file:JsModule("@emotion/hash")
```
Comment (turansky) [inline kotlin-emotion/src/main/kotlin/emotion/hash/hash.kt:1]: Do we need one more dependency?

**Review by turansky — CHANGES_REQUESTED** [review]

File: kotlin-emotion/src/main/kotlin/emotion/styled/styled.ext.kt (hunk 1 above)
Comment (sgrishchenko) [reply kotlin-emotion/src/main/kotlin/emotion/styled/styled.ext.kt:15]: `shouldForwardProp` can be useful

File: kotlin-emotion/src/main/kotlin/emotion/styled/generateId.kt (hunk 2 above)
Comment (sgrishchenko) [reply kotlin-emotion/src/main/kotlin/emotion/styled/generateId.kt:10]: It is removed

**Review by turansky — APPROVED** [review]

**Pull request merged**